### PR TITLE
Add `Distribution` properties to go along with `Release` info

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
From https://develop.sentry.dev/sdk/event-payloads/#optional-attributes

image

We had `release`, but not `dist` because it's usually used only in mobile apps.  For one example usage, see our [React Native docs](https://docs.sentry.io/platforms/react-native/sourcemaps/#using-custom-release-and-distribution).

We need this now, since we have the embedded Android and iOS SDKs.  However, for various reasons, it was much easier to make it an optional value available to all targets.

### CODE REVIEW

**Comment by mattjohnsonpint** [comment]
Env vars are leaking.  I'll fix soon.

File: src/Sentry/Internal/DistributionLocator.cs
Comment (SimonCropp) [inline src/Sentry/Internal/DistributionLocator.cs:5]: wouldnt this be simpler with a `GetEnvironmentVariable` in the static constructor and allow the resulting prop to be settable for testing purposes?

File: src/Sentry/Internal/MainSentryEventProcessor.cs
Comment (SimonCropp) [inline src/Sentry/Internal/MainSentryEventProcessor.cs:101]:
```
@event.Distribution ??= Distribution;
```

and the same on the two above?

File: src/Sentry/SentryOptions.cs
Comment (SimonCropp) [inline src/Sentry/SentryOptions.cs:230]: given this will appear in intellisense,  probs better to have an issue to track it and remove the todo?

File: src/Sentry/Internal/DistributionLocator.cs
Comment (mattjohnsonpint) [reply src/Sentry/Internal/DistributionLocator.cs:5]: It sure would.  I was copying the existing pattern from the other "locator" classes, but that would be much simpler.  I'll do that next.

Comment (mattjohnsonpint) [reply src/Sentry/Internal/DistributionLocator.cs:5]: We can do much better with testing on environment variables anyway, and those locators can be collapsed to a single class.

Comment (mattjohnsonpint) [reply src/Sentry/Internal/DistributionLocator.cs:5]: On second thought, I'm going to do that work in a separate PR first and then update this to use it later.  We'll hold this one for now.

**Comment by mattjohnsonpint** [comment]
This is ready.  Default values for release and distribution for Android and iOS will come in a separate PR after this is merged.

**Review by SimonCropp — APPROVED** [review]

**Pull request merged**
